pkg/store/proxy_test.go

```diff
@@ -2295,3 +2295,154 @@ func TestDedupRespHeap_Deduplication(t *testing.T) {
 	}
 
 }
+
+func TestDedupRespHeap_QuorumChunkDedup(t *testing.T) {
+	t.Parallel()
+
+	for _, tcase := range []struct {
+		responses []*storepb.SeriesResponse
+		testFn    func(responses []*storepb.SeriesResponse, h *responseDeduplicator)
+		tname     string
+	}{
+		{
+			tname: "edge case with only one response",
+			responses: []*storepb.SeriesResponse{
+				{
+					Result: &storepb.SeriesResponse_Series{
+						Series: &storepb.Series{
+							Labels: labelpb.ZLabelsFromPromLabels(labels.FromStrings("foo", "bar")),
+							Chunks: []storepb.AggrChunk{
+								{
+									Raw: &storepb.Chunk{
+										Type: storepb.Chunk_XOR,
+										Data: []byte(`abcdefgh`),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			testFn: func(responses []*storepb.SeriesResponse, h *responseDeduplicator) {
+				testutil.Equals(t, true, h.Next())
+				resp := h.At()
+				testutil.Equals(t, responses[0], resp)
+				testutil.Equals(t, false, h.Next())
+			},
+		},
+		{
+			tname: "keep 2 identical series",
```

Comment:
better to add a unit test for failure cases where there are 2 identical series + 1 corrupt series?

Reply:
Good idea. I added one.